core/trino-main/src/main/java/io/trino/execution/scheduler/SourcePartitionedScheduler.java

```diff
@@ -541,13 +540,13 @@ public void close()
         return newTasks.build();
     }
 
-    private Optional<RemoteTask> createTaskOnRandomNode()
+    private void createTaskOnRandomNode()
     {
         checkState(scheduledTasks.isEmpty(), "Stage task is already scheduled on node");
         List<InternalNode> allNodes = splitPlacementPolicy.allNodes();
         checkState(allNodes.size() > 0, "No nodes available");
         InternalNode node = allNodes.get(ThreadLocalRandom.current().nextInt(0, allNodes.size()));
-        return scheduleTask(node, ImmutableMultimap.of(), ImmutableMultimap.of());
+        scheduleTask(node, ImmutableMultimap.of(), ImmutableMultimap.of());
```

Comment:
In the old code this would return `Optional<RemoteTask>`.  Do we not need to handle the case where we got `Optional.empty()`?

Reply:
In old code this was used to return new tasks in `io.trino.execution.scheduler.ScheduleResult`. However, `io.trino.execution.scheduler.ScheduleResult#getNewTasks` is only used for testing purposes